operator/templates/node-scripts.yaml

```diff
@@ -4,10 +4,12 @@ metadata:
   name: {{ .Name }}-node-scripts
   namespace: {{ .Namespace }}
 data:
+  node-drain.sh: |
+    nodetool {{ if .Params.AUTHENTICATION_CREDENTIALS_SECRET }}-u $(cat /etc/cassandra/authentication/username) -pw $(cat /etc/cassandra/authentication/password){{ end }} drain
   node-readiness-probe.sh: |
-    nodetool status -p {{ .Params.JMX_PORT }} | grep -q "UN  ${POD_IP}"
+    nodetool {{ if .Params.AUTHENTICATION_CREDENTIALS_SECRET }}-u $(cat /etc/cassandra/authentication/username) -pw $(cat /etc/cassandra/authentication/password){{ end }} status -p {{ .Params.JMX_PORT }} | grep -q "UN  ${POD_IP}"
```

Comment:
agree with @ANeumann82

Reply:
I'll use a template variable instead of a shell variable. With this we don't have to repeat the variable definition for every command in the config map.